### PR TITLE
Error handling improvements

### PR DESCRIPTION
**Pull request by yavorm**
This PR proposes two changes:

1. Errors from the Sailthru API should throw exceptions to make error handling consistent. Otherwise the developer has to check for thrown exceptions as well as errors in the response.

2. Adds standard error codes to simplify error handling.

Thank you for reviewing, I'm looking forward to any feedback!

### CODE REVIEW

File: sailthru/Sailthru_Client_Exception.php [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
+    const ERROR_CODE_INVALID_EMAIL = 11;
 
-}
\ No newline at end of file
+    /**
+     * Standardized exception codes.
+     */
+    const CODE_GENERAL = 1000;
+    const CODE_RESPONSE_EMPTY = 1001;
+    const CODE_RESPONSE_INVALID = 1002;
```
Comment (yavorm) [inline sailthru/Sailthru_Client_Exception.php]: Sailthru currently uses error codes from 1 to 99. Starting at 1000 allows room for growth for custom error codes and avoids collision with any new error codes Sailthru may add to the API. I'm open to starting at a higher number too if you think it's a good idea.

Reference: https://getstarted.sailthru.com/new-for-developers-overview/api/api-response-errors/

**Comment by silverman63** [comment]
Thank you for the fix @yavorm. Merged.

**Comment by alucic** [comment]
@silverman63 this PR introduced a breaking change, and should have been tagged with major version bump. Unfortunately it was tagged as patch version (1.2.2 -> 1.2.3)

**Comment by silverman63** [comment]
@alucic agreed. I've been waiting to establish a proper release addressing a few other issues, which I'm going to try and get out this week. Apologize for any inconvenience it may have caused.

**Pull request merged**